### PR TITLE
feat(tinylicious-client): Promote APIs from `@beta` to `@public`

### PR DESCRIPTION
**Pull request by Josmithr**
The Tinylicious Client is used in public examples and documentation, and therefore requires public level support.

### CODE REVIEW

**Review by vladsud — APPROVED** [review]

**Comment by kashms** [comment]
Quick question: Did we find any additional info in the Git history for when this was made beta? I want to make sure we're not missing anything because going beta>public is easy but undoing it after a release will be a breaking change @Josmithr

**Comment by Josmithr** [comment]
> Quick question: Did we find any additional info in the Git history for when this was made beta? I want to make sure we're not missing anything because going beta>public is easy but undoing it after a release will be a breaking change @Josmithr

I was the person who made it beta. It had been `@internal` until I pointed out that it was used in examples and documentation and shouldn't be. At the time, FF API pushed to not make it public because we weren't sure if we wanted to encourage use of it. Beta was a sort of compromise to make it accessible. I argued at the time that it should be public, but was outvoted. But with our new discussion thread, there seems to be consensus that public is correct, especially since we don't seem to have consensus on whether or not we should _discourage_ its use.

File: packages/service-clients/tinylicious-client/api-report/tinylicious-client.alpha.api.md [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 export { CompatibilityMode }
 
-// @beta
+// @public
 export type ITinyliciousAudience = IServiceAudience<TinyliciousMember>;
 
-// @beta @sealed
+// @public @sealed
```
Comment (ChumpChief) [inline packages/service-clients/tinylicious-client/api-report/tinylicious-client.alpha.api.md:12]: FWIW, I think almost all of these interfaces could/should be `@sealed` - might be worth considering?

**Review by ChumpChief — APPROVED** [review]
Approved, but I'm not super up-to-speed on our usage of `@sealed` - might want to consider that as we make it public.

File: packages/service-clients/tinylicious-client/api-report/tinylicious-client.alpha.api.md (hunk 1 above)
Comment (Josmithr) [reply packages/service-clients/tinylicious-client/api-report/tinylicious-client.alpha.api.md:12]: I'll do a quick follow-up PR.

Comment (Josmithr) [reply packages/service-clients/tinylicious-client/api-report/tinylicious-client.alpha.api.md:12]: Actually, for interfaces like `User` and `Member`, I think we're probably okay with extension (correct me if I'm wrong). But the others probably make sense.

**Pull request merged**